File: Code/Core/Revenj.Security/UserPrincipal.cs
using System;
using System.Collections.Generic;
using System.Security.Principal;

namespace Revenj.Security
{
	internal class UserPrincipal : IPrincipal
	{
		private readonly IIdentity User;
		private readonly Lazy<HashSet<string>> Roles;

		public UserPrincipal(IIdentity user, Lazy<HashSet<string>> roles)
		{
			this.User = user;
			this.Roles = roles;
		}

		public IIdentity Identity { get { return User; } }

		public bool IsInRole(string role)
		{
			return role == Identity.Name || Roles.Value.Contains(role);
		}
	}
}

File: java/src/main/java/org/revenj/security/UserPrincipal.java
package org.revenj.security;

import java.security.Principal;
import java.util.Set;

public final class UserPrincipal implements Principal {

	private final String name;
	private final Set<String> roles;

	public UserPrincipal(String name) {
		this.name = name;
		this.roles = null;
	}

	public UserPrincipal(String name, Set<String> roles) {
		this.name = name;
		this.roles = roles;
	}

	@Override
	public String getName() {
		return name;
	}
}
